plugins/include/msgnatives.inc

```diff
@@ -0,0 +1,206 @@
+ *
+ * @noreturn
+ * @error               If "id" is non-zero and the client isn't
+ *                      connected, an error will be thrown.
+ */
+native hide_hud_elements(id, HideElemenentFlags:elements, bool:noadd = true, bool:reliable = true)
+
+/**
+ * Fades the client's screen.
+ *
+ * @param id            Client index or 0 for all clients
+ * @param duration      How long (in seconds) the fade is going to stay
+ *                      on the screen (0 - 16)
+ * @param fadetime      How many seconds is the fade going to fade in (0 - 16)
+ * @param flags         Screen fade flags:
+ *                        ScreenFade_FadeIn       - default
+ *                        ScreenFade_FadeOut      - fade out (not in)
+ *                        ScreenFade_Modulate     - modulate (don't blend)
+ *                        ScreenFade_StayOut      - ignores the duration and stays faded out until a new ScreenFade messages is received
+ * @param r             Red color amount (0 - 255)
+ * @param g             Green color amount (0 - 255)
+ * @param b             Blue color amount (0 - 255)
+ * @param a             Color alpha (brightness) (0 - 255)
+ * @param reliable      If true, the message will be sent via the reliable
+ *                      channel, otherwise it will use the unreliable one
+ *
+ * @noreturn
+ * @error               If "id" is non-zero and the client isn't
+ *                      connected, an error will be thrown.
+ */
+
+native fade_user_screen(id, Float:duration = 1.0, Float:fadetime = 0.5, ScreenFadeFlags:flags = ScreenFade_FadeIn, r = 0, g = 0, b = 255, a = 75, bool:reliable = true)
+
+/**
+ * Shakes the client's screen.
+ *
+ * @param id            Client index or 0 for all clients
+ * @param amplitude     Shake amplitude (0 - 16)
+ * @param duration      Shake duration (in seconds) (0 - 16)
+ * @param frequency     Delay between each shake (0 - 16)
+ * @param reliable      If true, the message will be sent via the reliable
+ *                      channel, otherwise it will use the unreliable one
+ *
+ * @noreturn
+ * @error               If "id" is non-zero and the client isn't
+ *                      connected, an error will be thrown.
+ */
+native shake_user_screen(id, Float:amplitude = 3.0, Float:duration = 3.0, Float:frequency = 1.0, bool:reliable = true)
+
+/**
+ * Changes the client's field of view (FOV).
+ *
+ * @note Setting the "fov" argument below 45 will also draw a sniper scope.
+ *
+ * @param id            Client index or 0 for all clients
+ * @param fov           Field of view degrees (0 - 255)
+ * @param reliable      If true, the message will be sent via the reliable
+ *                      channel, otherwise it will use the unreliable one
+ *
+ * @noreturn
+ * @error               If "id" is non-zero and the client isn't
+ *                      connected, an error will be thrown.
+ */
+native set_user_fov(id, fov = 0, bool:reliable = true)
```

Comment:
Why default `fov` is `0`?

Reply:
@WPMGPRoSToTeMa Because it resets the fov back to normal? Default values should do that in cases like this.

Reply:
@OciXCrom then maybe we need something like `reset_user_fov`?
And also why `user`, but not `player` or `client`? Function description says: "Changes the **_client_**'s field...".
There is also a mention of sniper scope, but that's not true for Half-Life.

Reply:
@WPMGPRoSToTeMa Take a look at the natives in the `fun` module - most of them have the reset value as default. For example, `set_user_gravity` has its `gravity` param set to `1.0` by default, so people can simply write `set_user_gravity(id)` in order to reset it. There's no need for people to research what the reset value for each native is before using it - let it be in the default value so they will know right away.

Reply:
@WPMGPRoSToTeMa I chose `user` for the same reason - respecting the `fun` module. It too has `user` in the native names and `client` in the description..

Reply:
I don't the others mod, but in CS, the default value is 90.

Reply:
@Arkshine 0 seems to have the same effect as 90.

Reply:
@OciXCrom is it `90` for all the games on GoldSrc? As I can see from the `SetFOV` client handler code the value `0` sets FOV to the default value, I hope this behavior is the same across the games. So it is better to stick with `0`.

Reply:
@WPMGPRoSToTeMa Okay, I switched it back to 0 again, if that's what the code says. There's no visible difference in CS between 0 and 90, so it shouldn't be a problem.